osler/core/models.py

```diff
@@ -352,12 +354,39 @@ def all_phones(self):
 
         return phones
 
+    def last_encounter(self):
+        if Encounter.objects.filter(patient=self.pk).exists():
+            last_encounter = Encounter.objects.filter(patient=self.pk).order_by('clinic_day').last()
+            return last_encounter
+        else:
+            return None
+
+    def get_status(self):
+        if self.last_encounter() is not None:
+            return self.last_encounter().status
+        else:
+            return default_inactive_status()
+
     def toggle_active_status(self, user, group):
         ''' Will Activate or Inactivate the Patient'''
-
         user_has_group = user.groups.filter(pk=group.pk).exists()
         if user_has_group and self.group_can_activate(group):
-            self.needs_workup = not self.needs_workup
+            #active encounter then inactivate
```

Comment:
This function is a real beast now. I wonder if we should break it out into a `patient.activate` and `patient.inactivate` function? I was never really sure about the whole toggle idea anyway...

Reply:
that just means that wherever you're calling it, you have to determine whether the patient is active or not and call the right function?

Reply:
Yeah, I guess that's why I did it this way initially. Just still feels odd. It's fine for now.